ocaml/xapi/xapi_periodic_scheduler.ml

```diff
@@ -30,9 +30,37 @@ let delay = Delay.make ()
 let (queue : (t Ipq.t)) = Ipq.create 50
 let lock = Mutex.create ()
 
+module Clock : sig
+  (** Oclock represents time in nanoseconds in a 64-bit integer. We
+   * can't make this abstract because module Ipq relies
+   * on it (for now). Note that the monotonic clock is only good for
+   * relative time measurements and the values returned by two different
+   * clocks like Oclock.monotonic and Oclock.realtime are unrelated. The
+   * underlying mechanism is clock_gettime(2).
+   *)
+  type t = Int64.t
+  val gettime:  unit -> t
+  val add:      t -> float -> t
+  val sub:      t -> t -> t
+  val to_sec:   t -> float
+end = struct
+  type t             = Int64.t
+  let clock          = Oclock.monotonic
+  let nano           = 1_000_000_000L (* 1E9 *)
```

Comment:
I know it's only an implementation detail internal to this Clock module, but this looks to me like `giga` or `inv_nano`: I would expect `nano` to be `1E-9`.